youtube_dl/extractor/sverigesradio.py

```diff
@@ -0,0 +1,69 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..compat import compat_str
+from ..utils import (
+    try_get,
+    int_or_none,
+)
+
+
+class SverigesRadioBaseIE(InfoExtractor):
+    _GEO_COUNTRIES = ['SE']
+
+    def _extract_audio(self, audio_type, audio_id):
+        query = {'id': audio_id, 'type': audio_type}
+        audiourl = self._download_json('https://sverigesradio.se/sida/playerajax/getaudiourl', audio_id, query=dict(query, quality='high', format='iis'))
+        metadata = self._download_json('https://sverigesradio.se/sida/playerajax/audiometadata', audio_id, query=query)
+        item = metadata['items'][0]
+
+        return {
+            'id': compat_str(audio_id),
+            'url': compat_str(audiourl['audioUrl']),
+            'title': compat_str(item['subtitle']),
+            'series': try_get(item, lambda x: x['title'], compat_str),
+            'duration': int_or_none(item.get('duration')),
+            'thumbnail': try_get(item, lambda x: x['displayimageurl'], compat_str),
+            'description': try_get(item, lambda x: x['description'], compat_str)
+        }
+
+
+class SverigesRadioArtikelIE(SverigesRadioBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?sverigesradio\.se/sida/artikel.aspx\?.*artikel=(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'https://sverigesradio.se/sida/artikel.aspx?programid=83&artikel=7038546',
+        'md5': '6a4917e1923fccb080e5a206a5afa542',
+        'info_dict': {
+            'id': '7038546',
+            'ext': 'm4a',
+            'duration': 132,
+            'series': 'Nyheter (Ekot)',
+            'title': 'Esa Teittinen: Sanningen har inte kommit fram',
+            'description': 'md5:daf7ce66a8f0a53d5465a5984d3839df',
+            'thumbnail': 're:^https://static-cdn.sr.se/sida/images/'
+        }
+    }
+
+    def _real_extract(self, url):
+        return self._extract_audio('publication', self._match_id(url))
+
+
+class SverigesRadioAvsnittIE(SverigesRadioBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?sverigesradio\.se/(?:sida/)?avsnitt/(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'https://sverigesradio.se/avsnitt/1140922?programid=1300',
+        'md5': '20dc4d8db24228f846be390b0c59a07c',
+        'info_dict': {
+            'id': '1140922',
+            'ext': 'mp3',
+            'duration': 3307,
+            'series': 'Konflikt',
+            'title': 'Metoo och valen',
+            'description': 'md5:fcb5c1f667f00badcc702b196f10a27e',
+            'thumbnail': 're:^https://static-cdn.sr.se/sida/images/'
+        }
+    }
+
+    def _real_extract(self, url):
+        return self._extract_audio('episode', self._match_id(url))
```

Comment:
Another format is available at https://sverigesradio.se/sida/playerajax/getaudiourl?id=episode_id&type=secondary&quality=high&format=iis.

Reply:
Do you have an example of an id what works with `type=secondary`? i get `{"audioUrl":"","duration":0,"codingFormat":-1,"state":101,"isGeoblockEnabled":false}` when i try a couple of ones.

Should i test both types and return and array with valid infos?

Reply:
Found one `https://sverigesradio.se/sida/playerajax/getaudiourl?id=1233546&type=secondary&quality=high&format=iis`